assets/ca/ca.querytranslator.js

```diff
@@ -0,0 +1,387 @@
+     */
+	escapeValue = function (value) {
+		return value.replace(/([\-\+&\|!\(\)\{}\[\]\^"~\*\?:\\])/, '\\$1');
+	};
+
+	/**
+	 * Convert a set of rules from the jQuery query builder into a CA search query.  Performs the inverse operation to
+	 * `convertSearchQueryToQueryBuilderRules`.
+	 * @param {Object} ruleSet
+	 * @returns {String}
+	 */
+	caUI.convertQueryBuilderRuleSetToSearchQuery = function (ruleSet) {
+		if (ruleSet.condition && ruleSet.rules) {
+			return '(' + $.map(ruleSet.rules, caUI.convertQueryBuilderRuleSetToSearchQuery).join(' ' + ruleSet.condition + ' ') + ')';
+		}
+		if (ruleSet.operator && ruleSet.field) {
+			// Escape value to allow special characters
+			var negation = ruleSet.operator.match(/not_/),
+				prefix = ruleSet.field + (negation ? ':-' : ':');
+			switch (negation ? ruleSet.operator.replace('not_', '') : ruleSet.operator) {
+				case 'equal':
+					return prefix + '"' + escapeValue(ruleSet.value) + '"';
+				case 'in':
+					return prefix + '(' + escapeValue(ruleSet.value) + ')';
+				case 'between':
+					return prefix + '[' + escapeValue(ruleSet.value[0]) + ' TO ' + escapeValue(ruleSet.value[1]) + ']';
+				case 'begins_with':
+					return prefix + '"' + escapeValue(ruleSet.value) + '*"';
+				case 'contains':
+					return prefix + '"*' + escapeValue(ruleSet.value) + '*"';
+				case 'ends_with':
+					return prefix + '"*' + escapeValue(ruleSet.value) + '"';
+				case 'is_empty':
+				case 'is_null':
+					return prefix + '*';
+			}
+			return ruleSet.field + ':' + ruleSet.value;
+		}
+		return '';
+	};
+
+	/**
+	 * Retrieve the list of tokens from the given query string.
+	 * @param {String} query
+	 * @return {Array}
+     */
+	getTokenList = function (query) {
+		var token,
+			tokens = [],
+			queryArray = query.trim().split('');
+		while (token = shiftToken(queryArray)) {
+			tokens.push(token);
+		}
+		while (tokens.length >= 2 && tokens[0].type === TOKEN_LPAREN && tokens[tokens.length - 1].type === TOKEN_RPAREN) {
+			tokens = tokens.slice(1, tokens.length - 1);
+		}
+		return tokens;
+	};
+
+	/**
+	 * Retrieve the next token from the `queryArray`, which is destructively processed.
+	 * @param {Array} queryArray
+	 * @returns {Object|undefined}
+	 */
+	shiftToken = function (queryArray) {
+		var character, token,
+			quoted = false,
+			escaped = false,
+			end = false;
+		// End condition.
+		if (queryArray.length === 0) {
+			return undefined;
+		}
+		// Inspect the first character to determine the type of token.
+		character = queryArray.shift();
+		switch (character) {
+			// Single-character tokens.
+			case '(':
+				token = { type: TOKEN_LPAREN };
+				end = true;
+				break;
+			case ')':
+				token = { type: TOKEN_RPAREN };
+				end = true;
+				break;
+			case ':':
+				token = { type: TOKEN_COLON };
+				end = true;
+				break;
+			case '-':
+				token = { type: TOKEN_NEGATION };
+				end = true;
+				break;
+			// Collapse whitespace.
+			case ' ':
+				while (queryArray[0] === ' ') {
```

Comment:
Yes I think both of these cases are true. I got the impression that hiding the query builder disabled it, so I think yes if it's hidden it should be disabled, and an invalid search should not clear the box, but you should somehow notify the user where it went wrong. jQuery growl maybe? (or was it toast - I can't remember the one that CA uses).